File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n=nums.size();
        vector<int>freqarr(n+1,0);
        for(int i=0;i<n;i++){
            freqarr[nums[i]]++;
        }
        int res=0;
        for(int i=0;i<freqarr.size();i++){
            if(freqarr[i]>=2){
                res=i;
            }
        }
        return res;
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int res=0;
        Map<Integer,Integer>map=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        for(int i=0;i<nums.length;i++){
            if(map.get(nums[i])>=2){
                res=nums[i];
            }
        }
        return res;
    }
}